proxy/proxy.go

```diff
@@ -31,3 +34,12 @@ func MakeHTTPClient(timeout *time.Duration) http.Client {
 	// This should be used for faas-cli invoke etc.
 	return http.Client{}
 }
+
+// MakeHTTPRequest makes HTTP requests with built-in user-agent string
+func AddUserAgent(req *http.Request) {
+	req.Header.Set("User-Agent", getUserAgent())
```

Comment:
Not sure about this mutable/destructive operation. I guess as a nit I'd probably have done it as:

```
req.Header.Set("User-Agent", version.getUserAgent())
```

Without a function that takes a http.Request.

Reply:
That would lead to multiple places to test a generic function though right? That's why I did it like I did, test one place in isolation without needing to worry about openfaas-specifics.